External PI packages cannot resolve @mariozechner/* imports (module aliasing needed)

## Problem

When users install external PI ecosystem packages (e.g., `pi-rtk`, `pi-context`, `pi-agent-browser`, etc.) via `settings.json`, these packages fail to load with module resolution errors:

```
[gsd] Extension load error: Failed to load extension: Cannot find module '@mariozechner/pi-coding-agent'
Require stack:
- /opt/homebrew/lib/node_modules/pi-rtk/index.ts
[gsd] Extension load error: Failed to load extension: Cannot find module '@mariozechner/pi-ai'
Require stack:
- /Users/alex/.gsd/agent/extensions/github/index.ts
```

### Root Cause

1. The PI ecosystem packages on npm are built for the original `pi` agent and import from `@mariozechner/pi-*` scope
2. GSD forked these packages to `@gsd/pi-*` scope
3. External packages still have imports like `import { ExtensionAPI } from "@mariozechner/pi-coding-agent"`
4. These imports fail because GSD's `node_modules` only contains `@gsd/*` packages, not `@mariozechner/*`

### My Setup

**settings.json:**
```json
{
  "packages": [
    "npm:pi-rtk",
    "npm:pi-markdown-preview",
    "npm:pi-librarian",
    "npm:pi-context",
    "npm:pi-perplexity",
    "npm:pi-agent-browser",
    "npm:pi-agentic-compaction",
    "npm:pi-superpowers-plus"
  ]
}
```

These packages are installed globally by npm and loaded by GSD at runtime.

## Current Workaround

I manually created wrapper `package.json` files that alias `@mariozechner/*` → `@gsd/*`:

```bash
# Create wrapper packages in gsd-pi's node_modules
mkdir -p /opt/homebrew/lib/node_modules/gsd-pi/node_modules/@mariozechner/pi-coding-agent
cat > /opt/homebrew/lib/node_modules/gsd-pi/node_modules/@mariozechner/pi-coding-agent/package.json << 'PKG'
{
  "name": "@mariozechner/pi-coding-agent",
  "version": "0.57.1",
  "type": "module",
  "main": "../../@gsd/pi-coding-agent/dist/index.js",
  "types": "../../@gsd/pi-coding-agent/dist/index.d.ts"
}
PKG
# Repeat for pi-ai, pi-tui, pi-agent-core...
```

This works because Node's `main` field allows `../` paths (unlike `exports`).

### Workaround Limitations

- **Fragile**: Breaks on every `npm install -g gsd-pi` update
- **Manual**: Must be reapplied after each update
- **Incomplete**: Need to create wrappers for all 4 core packages
- **Not discoverable**: Users hitting this error won't know the fix

## Suggested Permanent Fixes

### Option 1: Module Aliasing in loader.ts (Recommended)

Add package aliasing in `loader.ts` using Node's `Module._initPaths()` or a custom resolver. The loader already sets `NODE_PATH` for dependency resolution - extend it to handle package name aliasing:

```typescript
// In loader.ts, after setting NODE_PATH
// Create alias packages for @mariozechner/* → @gsd/*
const aliasDir = join(gsdNodeModules, '@mariozechner')
const aliases = ['pi-coding-agent', 'pi-ai', 'pi-tui', 'pi-agent-core']

for (const pkg of aliases) {
  const aliasPath = join(aliasDir, pkg)
  const targetPath = join(gsdNodeModules, '@gsd', pkg)
  
  if (existsSync(targetPath) && !existsSync(aliasPath)) {
    mkdirSync(aliasPath, { recursive: true })
    writeFileSync(join(aliasPath, 'package.json'), JSON.stringify({
      name: `@mariozechner/${pkg}`,
      version: '0.57.1', // could read from target
      type: 'module',
      main: `../../@gsd/${pkg}/dist/index.js`,
      types: `../../@gsd/${pkg}/dist/index.d.ts`
    }, null, 2))
  }
}
```

### Option 2: Postinstall Script

Add a `postinstall` script that creates the alias packages. This is simpler but requires write access during install.

### Option 3: Jiti Alias Configuration

If jiti supports module aliases, configure it to map `@mariozechner/*` → `@gsd/*` at the loader level.

### Option 4: Symlinks (Alternative)

Create symlinks instead of wrapper files:
```bash
ln -s ../@gsd/pi-coding-agent @mariozechner/pi-coding-agent
```
Note: Symlinks work but may have issues with some tools that don't follow them properly.

## Impact

This affects any user who wants to use the rich ecosystem of PI packages (pi-rtk, pi-context, pi-agent-browser, etc.) with GSD. Currently, these packages are unusable without manual workarounds.

## Related

- Issue #155 (different cause - missing @gsd/pi-coding-agent package)

---

**Environment:**
- GSD version: 2.7.0
- Node version: v24.x
- OS: macOS (Homebrew install)
- Install method: `npm install -g gsd-pi`

--- a/packages/pi-coding-agent/src/core/extensions/loader.ts
+++ b/packages/pi-coding-agent/src/core/extensions/loader.ts
@@ -46,6 +46,12 @@ const VIRTUAL_MODULES: Record<string, unknown> = {
 	"@gsd/pi-ai": _bundledPiAi,
 	"@gsd/pi-ai/oauth": _bundledPiAiOauth,
 	"@gsd/pi-coding-agent": _bundledPiCodingAgent,
+	// Aliases for external PI ecosystem packages that import from the original scope
+	"@mariozechner/pi-agent-core": _bundledPiAgentCore,
+	"@mariozechner/pi-tui": _bundledPiTui,
+	"@mariozechner/pi-ai": _bundledPiAi,
+	"@mariozechner/pi-ai/oauth": _bundledPiAiOauth,
+	"@mariozechner/pi-coding-agent": _bundledPiCodingAgent,
 };
 
 const require = createRequire(import.meta.url);
@@ -80,6 +86,12 @@ function getAliases(): Record<string, string> {
 		"@gsd/pi-ai": resolveWorkspaceOrImport("ai/dist/index.js", "@gsd/pi-ai"),
 		"@gsd/pi-ai/oauth": resolveWorkspaceOrImport("ai/dist/oauth.js", "@gsd/pi-ai/oauth"),
 		"@sinclair/typebox": typeboxRoot,
+		// Aliases for external PI ecosystem packages that import from the original scope
+		"@mariozechner/pi-coding-agent": packageIndex,
+		"@mariozechner/pi-agent-core": resolveWorkspaceOrImport("agent/dist/index.js", "@gsd/pi-agent-core"),
+		"@mariozechner/pi-tui": resolveWorkspaceOrImport("tui/dist/index.js", "@gsd/pi-tui"),
+		"@mariozechner/pi-ai": resolveWorkspaceOrImport("ai/dist/index.js", "@gsd/pi-ai"),
+		"@mariozechner/pi-ai/oauth": resolveWorkspaceOrImport("ai/dist/oauth.js", "@gsd/pi-ai/oauth"),
 	};
 
 	return _aliases;

